File: FragmentIntroduction/app/src/main/java/com/akashsoam/fragmentintroduction/MainActivity.java
package com.akashsoam.fragmentintroduction;

import androidx.appcompat.app.AppCompatActivity;
import androidx.fragment.app.FragmentManager;
import androidx.fragment.app.FragmentTransaction;

import android.os.Bundle;
import android.widget.FrameLayout;
import android.widget.Toast;

public class MainActivity extends AppCompatActivity implements AndroidVersionInterface{
    private FrameLayout fragmentContainer;
    FragmentManager fragmentManager;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        fragmentContainer = findViewById(R.id.fragment_container);



        fragmentManager = getSupportFragmentManager();
        FragmentTransaction fragmentTransaction = fragmentManager.beginTransaction();


        DemoFragment demoFragment = new DemoFragment();
        fragmentTransaction.add(fragmentContainer.getId(), demoFragment);
        fragmentTransaction.commit();

    }

    @Override
    public void onVersionItemClick(String versionName) {
        Toast.makeText(this, "Clicked "+versionName, Toast.LENGTH_SHORT).show();

        FragmentTransaction fragmentTransaction = fragmentManager.beginTransaction();

        Bundle bundle = new Bundle();
        bundle.putString("versionName", versionName);

        DetailFragment detailFragment = new DetailFragment();
        detailFragment.setArguments(bundle);
        fragmentTransaction.replace(R.id.fragment_detail_container,detailFragment);
        fragmentTransaction.commit();
    }
}

File: FirestoreLearn/app/src/main/java/com/akashsoam/firestorelearn/MainActivity.kt
package com.akashsoam.firestorelearn

import android.os.Bundle
import android.widget.Button
import android.widget.EditText
import android.widget.TextView
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import com.google.firebase.firestore.DocumentReference
import com.google.firebase.firestore.FirebaseFirestore

class MainActivity : AppCompatActivity() {

    private lateinit var editTextTitle: EditText
    private lateinit var editTextDescription: EditText
    private lateinit var saveButton: Button
    private lateinit var loadButton: Button
    private lateinit var textViewData: TextView

    private val db: FirebaseFirestore = FirebaseFirestore.getInstance()
    private val docRef: DocumentReference = db.collection("Notebook").document("My first note")
    private val KEY_TITLE = "title"
    private val KEY_DESCRIPTION = "description"

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        editTextTitle = findViewById(R.id.edit_text_title)
        editTextDescription = findViewById(R.id.edit_text_description)
        saveButton = findViewById(R.id.button_save)
        loadButton = findViewById(R.id.button_load)

        saveButton.setOnClickListener {
            saveNote()
        }
        loadButton.setOnClickListener {
            loadData()
        }
    }


    private fun saveNote() {
        val title = editTextTitle.text.toString()
        val description = editTextDescription.text.toString()

        val note = mutableMapOf<String, Any>()
        note[KEY_TITLE] = title
        note[KEY_DESCRIPTION] = description

        docRef.set(note).addOnSuccessListener {
            Toast.makeText(
                this@MainActivity,
                "note added successfully!",
                Toast.LENGTH_SHORT
            ).show()
        }.addOnFailureListener {
            Toast.makeText(
                this@MainActivity,
                "Could NOT add the note",
                Toast.LENGTH_SHORT
            ).show()
        }
    }

    private fun loadData() {
        docRef.get().addOnSuccessListener { document ->

            val title = document.getString(KEY_TITLE)
            val description = document.getString(KEY_DESCRIPTION)
            val note = mutableMapOf<String, Any>()
            note["title"] = title!!
            note["description"] = description!!
        }.addOnFailureListener {
            Toast.makeText(this@MainActivity, "retrieval NOT successfull!", Toast.LENGTH_SHORT)
                .show()
        }
    }


}